index-configuration-tool/main.py

```diff
@@ -0,0 +1,135 @@
+    # Remove internal settings
+    result = dict(resp.json())
+    for index in result:
+        for setting in INTERNAL_SETTINGS_KEYS:
+            result[index][SETTINGS_KEY][INDEX_KEY].pop(setting, None)
+    return result
+
+
+def validate_logstash_config(config: dict):
+    if "input" not in config or "output" not in config:
+        raise ValueError("Missing input or output data from Logstash config")
+    for plugin_tuple in config["input"], config["output"]:
+        input_type = plugin_tuple[0]
+        if input_type not in SUPPORTED_ENDPOINTS:
+            raise ValueError("Unsupported plugin type: " + input_type)
+        plugin_config = plugin_tuple[1]
+        if HOSTS_KEY not in plugin_config:
+            raise ValueError("No hosts defined for plugin: " + input_type)
+        if USER_KEY in plugin_config and PWD_KEY not in plugin_config:
+            raise ValueError("Invalid auth configuration (no password for user) for plugin: " + input_type)
+        elif PWD_KEY in plugin_config and USER_KEY not in plugin_config:
+            raise ValueError("Invalid auth configuration (Password without user) for plugin: " + input_type)
+
+
+def has_differences(index: str, dict1: dict, dict2: dict, key: str) -> bool:
+    if key in dict1[index] and key in dict2[index]:
+        data_diff = diff(dict1[index][key], dict2[index][key])
+        return bool(data_diff)
+    else:
+        return True
+
+
+def print_report(source: dict, target: dict) -> set:
+    index_conflicts = set()
+    indices_in_target = set(source.keys()) & set(target.keys())
+    for index in indices_in_target:
+        # Check settings
+        if has_differences(index, source, target, SETTINGS_KEY):
+            index_conflicts.add(index)
+        # Check mappings
+        if has_differences(index, source, target, MAPPINGS_KEY):
+            index_conflicts.add(index)
+    identical_indices = set(indices_in_target) - set(index_conflicts)
+    indices_to_create = set(source.keys()) - set(indices_in_target)
+    # Print report
+    print("Indices in target cluster with conflicting settings/mappings: " + string_from_set(index_conflicts))
+    print("Indices already created in target cluster (data will NOT be moved): " + string_from_set(identical_indices))
+    print("Indices to create: " + string_from_set(indices_to_create))
+    return indices_to_create
+
+
+def create_indices(indices: dict, endpoint: str, auth_tuple: tuple):
+    for index in indices:
+        actual_endpoint = endpoint + index
+        data_dict = dict()
+        data_dict[SETTINGS_KEY] = indices[index][SETTINGS_KEY]
+        data_dict[MAPPINGS_KEY] = indices[index][MAPPINGS_KEY]
+        try:
+            resp = requests.put(actual_endpoint, auth=auth_tuple, json=data_dict)
+            resp.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            print("Failed to create index [" + index + "] - " + str(e), file=sys.stderr)
```

Comment:
minor, but f-string formatting is really nice for making this sort of thing easier to read:
```
print(f"Failed to create index [{index}] - {e}", file=sys.stderr)
```